canonical/cloud-init
# Added skip_detect_openstack option for OpenStack

## Added "skip_detect_openstack" for OpenStack datasource

```
summary: Fixing an issue when running the OpenStack datasource on Bare Metal machines

LP: #1815990 (might be fixed as well)
```

### Please squash commits on merge

## Additional Context
Running CloudInit with the OpenStack datasource configured on a Bare Metal machine is resulting in an issue since the DataSource itself also checks if it's running on OpenStack inside a virtual machine using dmi decode results. 

Maybe fixes a 2019 issue or at least gives an workaround for it https://bugs.launchpad.net/cloud-init/+bug/1815990

## Test Steps
Removing that check allows to run the OpenStack data source also on bare metal provisioned machines. I have verified that Cloud Init executes on Bare Metal now and if the default still behaves the same.

## Checklist:
